docs/24-efficiency-and-tastefulness.markdown

```diff
@@ -0,0 +1,300 @@
+# Chapter 24. Efficiency and Tastefulness
+
+## 24.1. Efficiency
+
+Actually, you make MDL programs efficient by thinking hard about what
+they really make the interpreter **do**, and making them do less.
+Some guidelines, in order of decreasing expense:
+
+1. Free storage is expensive.
+2. Calling functions is expensive.
+3. `PROG` and `REPEAT` are expensive, except when compiled.
+
+Explanation:
+
+1. Unnecessary use of free storage (creating needless `LIST`s,
+`VECTOR`s, `UVECTOR`s, etc.) will cause the garbage collector to run
+more often. This is **expensive!!** A fairly large MDL (for example,
+60,000 36-bit words) can take ten seconds of PDP-10 CPU time to
```

Comment:
to garbage-collect -> for a garbage collection

side note: 60,000 doesn't appear to have a comma in my PDF, but I'm inclined to keep it anyway since this is US English